### PR TITLE
add coq-action.yml

### PR DESCRIPTION
**Pull request by caldwellb**
Migrating the CI used in other inQWIRE projects

### CODE REVIEW

**Comment by khieta** [comment]
Super excited for CI for this repo!! Might be useful to merge my PR (#51) before debugging why the this build is failing, since my PR will likely break it even worse 😄

**Comment by caldwellb** [comment]
> Super excited for CI for this repo!! Might be useful to merge my PR (#51) before debugging why this build is failing, since my PR will likely break it even worse smile

Thanks for the heads up! Will definitely wait until #51 is merged before I return to debug this.

**Review by adrianleh — APPROVED** [review]
CI looks to be working as intended. Approved

**Pull request merged**
